packages/amplify_datastore_plugin_interface/lib/src/types/query/query_predicate.dart

```diff
@@ -1,134 +1,162 @@
-
-  // and
-  QueryPredicateGroup and(QueryPredicate predicate) {
-    if (this.type == QueryPredicateGroupType.and) {
-      predicates.add(predicate);
-      return this;
-    }
-    return new QueryPredicateGroup(
-        QueryPredicateGroupType.and, [this, predicate]);
-  }
-
-  QueryPredicateGroup operator &(QueryPredicate predicate) => and(predicate);
-
-  // or
-  QueryPredicateGroup or(QueryPredicate predicate) {
-    if (this.type == QueryPredicateGroupType.or) {
-      predicates.add(predicate);
-      return this;
-    }
-    return new QueryPredicateGroup(
-        QueryPredicateGroupType.or, [this, predicate]);
-  }
-
-  QueryPredicateGroup operator |(QueryPredicate predicate) => or(predicate);
-
-  @override
-  bool evaluate(Model model) {
-    switch (type) {
-      case QueryPredicateGroupType.and:
-        return predicates.every((predicate) => predicate.evaluate(model));
-      case QueryPredicateGroupType.or:
-        return predicates.any((predicate) => predicate.evaluate(model));
-      case QueryPredicateGroupType.not:
-        return predicates.every((predicate) => !predicate.evaluate(model));
-    }
-  }
-
-  @override
-  Map<String, dynamic> serializeAsMap() {
-    return <String, dynamic>{
-      'queryPredicateGroup': <String, dynamic>{
-        'type': type.toShortString(),
-        'predicates':
-            predicates.map((predicate) => predicate.serializeAsMap()).toList()
-      },
-    };
-  }
-}
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+part of 'query_field.dart';
+
+// Top level global function to be used without the context of the enclosing class
+QueryPredicateGroup not(QueryPredicate predicate) {
+  return new QueryPredicateGroup(QueryPredicateGroupType.not, [predicate]);
+}
+
+/// Represents individual conditions or groups of conditions
+/// that are used to query data
+abstract class QueryPredicate<T extends Model> {
+  static final _QueryPredicateAll =
+      _QueryPredicateConstant(QueryPredicateConstantType.all);
+
+  const QueryPredicate();
+
+  /// A static instance of [QueryPredicate] that matches any object.
+  ///
+  /// You can use [QueryPredicate.all] with [DataStoreSyncExpression] like this:
+  /// ```dart
+  /// // A sync expression matching any object of Post
+  /// var syncExpression = DataStoreSyncExpresion(
+  ///   Post.classType,
+  ///   () => QueryPredicate.all,
+  /// )
+  /// ```
+  static QueryPredicate get all => _QueryPredicateAll;
```

Comment:
*nit:* Couldn't you remove `_QueryPredicateAll` above and update this to `static QueryPredicate get all => _QueryPredicateConstant(QueryPredicateConstantType.all);`

Reply:
Wouldn't that return a new instance on every `QueryPredicate.all` call? This instance can be reused as it is not model specific.